packages/synapse-react-client/src/utils/AppUtils/AppUtils.ts

```diff
@@ -39,17 +39,24 @@ export function processRedirectURLInOneSage() {
   // PORTALS-3299 : Indicate that we have completed the login workflow (cookie expires in a minute) to break out of a cycle
   const expireDate = new Date()
   expireDate.setMinutes(expireDate.getMinutes() + 1)
-  const hostname = window.location.hostname.toLowerCase()
   cookies.set(ACCOUNT_SITE_PROMPTED_FOR_LOGIN_COOKIE_KEY, 'true', {
     path: '/',
     expires: expireDate,
-    domain: hostname.endsWith('.synapse.org') ? 'synapse.org' : undefined,
+    domain: getCookieDomain(),
   })
 
   if (cookies.get(ONE_SAGE_REDIRECT_COOKIE_KEY)) {
     const href = cookies.get(ONE_SAGE_REDIRECT_COOKIE_KEY)
-    cookies.remove(ONE_SAGE_REDIRECT_COOKIE_KEY)
-    window.location.assign(href)
+    // instead of removing, set the expiration to 10 seconds to avoid race condition with SageAccountWeb LoggedInRedirector
+    const tenSecondsFromNow = new Date()
```

Comment:
This was a tricky bug, and it may only be happening in react strict mode (double render), but the LoggedInRedirector was calling the window.location.replace() (with the Source Config app URL) in parallel to this method calling window.location.assign() (with the cookie value that contains the One Sage redirect URL).

Reply:
Before this change, the workflow would usually (in my environment) send me to Synapse.org (the source app URL) rather than the URL stored in the cookie.  
The request to get the correct ONE_SAGE_REDIRECT_COOKIE_KEY value was in the network activity, but reported (in Chrome) as "cancelled".

Reply:
Should we have a unique appId for prod OAuth signin?

Reply:
What do you think of having a dedicated route on OneSage (`/oauth` ?) that we redirect to from the OAuth sign in app, that doesn't use the LoginRedirector?

Reply:
It's a good idea.  Would we also need to make updates to handle SSO (Google and ORCiD) login (update the supported redirect endpoints in those apps)?

Reply:
Wait, the LoggedInRedirector is the one responsible for processing the redirect cookie (or falling back to redirecting based on the Source App redirect URL).  I suppose we could write a new LoggedInRedirector that never attempts to use the appContext.redirectURL.  Is this what you were envisioning?

Reply:
A different component similar to LoggedInRedirector was what I had in mind. I think you might be right about the SSO issues--but maybe that's not a big deal.

Another idea is to add appIds for the production oauth signin app (+ one for each environment) and treat it like any other app--would that work with the OAuth search parameters?

Reply:
Search parameters are currently not stored/restored on app redirect